### PR TITLE
Allow custom pacer/BWE to be more piecemeal

### PR DESCRIPTION
**Pull request by pthatcher**

### CODE REVIEW

File: explainer-use-case-2.md [hunk 1]
```diff
@@ -52,6 +52,11 @@ enum RtpUnsentReason {
 partial interface RtpTransport {
   attribute EventHandler onrtpsent;  // RtpSent
   attribute EventHandler onrtpacksreceived;  // RtpAcks
+  attribute unsigned long customMaxBandwidth;
+
+  attribute bool customPacer;
```
Comment (tonyherre) [inline explainer-use-case-2.md:57]: What does having a specific `customPacer` field give us over just seeing whether `sendTime` is set in the sendRtp() calls?

File: explainer-use-case-2.md [hunk 2]
```diff
@@ -52,6 +52,11 @@ enum RtpUnsentReason {
...
+  attribute EventHandler onrtppacketized;  // No payload.  Call readPacketizedRtp
```
Comment (tonyherre) [inline explainer-use-case-2.md:58]: `onrtppacketsavailable` might make it clearer there's nothing in the event? onrtppacketized is fine for me too, so feel free to ignore.

File: explainer-use-case-2.md [hunk 3]
```diff
@@ -52,6 +52,11 @@ enum RtpUnsentReason {
...
+  sequence<RtpPacket> readPacketizedRtp(maxNumberOfPackets);
```
Comment (tonyherre) [inline explainer-use-case-2.md:59]: The requirements here don't actually need the full RtpPacket objects created by the UA to be created and passed up, just the sending to be controlled, so I'd be keen on us discussing alternative shapes here - particularly ones which can address the performance concerns in #20.

File: explainer-use-case-2.md [hunk 4]
```diff
@@ -102,10 +107,15 @@ rtpTransport.onrtpacksreceived = (rtpAcks) => {
 ## Example 2: Custom Pacing and Probing
 
 ```javascript
-const [pc, rtpSender1, rtpSender2] = setupPeerConnectionWithRtpSenders();  // Custom
+const [pc, rtpTransport] = setupPeerConnectionWithRtpTransport();  // Custom
 const pacer = createPacer();  // Custom
+rtpTransport.customPacer = true;
+rtpTransport.onrtppacketized = () => {
```
Comment (tonyherre) [inline explainer-use-case-2.md:113]: Let's add an example here which makes use of the event/readPacketizedRtp() separation. I can take a stab at a PR.

**Review by tonyherre — COMMENTED** [review]
Nice! Thanks for writing up

File: explainer-use-case-2.md (hunk 4 above)
Comment (tonyherre) [reply explainer-use-case-2.md:113]: https://github.com/w3c/webrtc-rtptransport/pull/25 - WDYT?

File: explainer-use-case-2.md (hunk 2 above)
Comment (pthatcher) [reply explainer-use-case-2.md:58]: I like the idea

File: explainer-use-case-2.md (hunk 3 above)
Comment (pthatcher) [reply explainer-use-case-2.md:59]: But you'd want some of the metadata data, would you not?  Like SSRC, seqnum, PR, timestamp, if you are building a pacer that is smart about what to drop when?  For that matter, you might want to inspect the payload.

File: explainer-use-case-2.md (hunk 2 above)
Comment (pthatcher) [reply explainer-use-case-2.md]: Change to onpacketizedrtpavailable

**Pull request merged**